torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -21,14 +23,28 @@ def __init__(
         self,
         data,
         *args,
+        edge_time=None,
         neg_sampling_ratio: float = 0.0,
         num_src_nodes: Optional[int] = None,
         num_dst_nodes: Optional[int] = None,
         **kwargs,
     ):
         super().__init__(data, *args, **kwargs)
         self.neg_sampling_ratio = neg_sampling_ratio
-
+        self.edge_time = edge_time
+
+        # TODO if self.edge_time is not None and
+        # input_type doesn't have time attribute
+        # set it to float(inf).
+
+        if self.edge_time is not None:
```

Comment:
Not really sure why we need to do this in `__init__`. Shouldn't we want to do this within `__call__`? I assume for this we can modify `_hetero_sparse_neighbor_sample` to take in additional attributes.

Reply:
I agree with this, we are having to infer during `__call__` whether there is an `edge_time` present in the data sampled from the dataset anyway.